common/persistence/client/targeted_fault_injection.go

```diff
@@ -100,25 +100,85 @@ func (d *dataStoreErrorGenerator) Generate() error {
 
 // newError returns an error based on the provided name. If the name is not recognized, then this method will
 // panic.
-func newError(errName string, errRate float64, methodName string) error {
+
+type fault struct {
+	err error
+	// execOp indicates whether the operation should be executed before returning the error.
+	execOp bool
+}
+
+func newFault(err error) *fault {
+	return &fault{
+		err:    err,
+		execOp: false,
+	}
+}
+func inject0(f *fault, op func() error) error {
```

Comment:
You can probably do this dynamically using reflection (get its type and use `.Out(i)` and `.In(i)`)

Reply:
I considered it, but given the fact that there is just 3 of them and `inject2` is used only once, I decided to go with this route. I also don't want to introduce extra delay here. And if you don't like those `inject*` names, look at [this](https://go.dev/wiki/RangefuncExperiment), it is mainstream now:-)